ui/utils/format.ts

```diff
@@ -0,0 +1,17 @@
+const K = 1024;
+const UNITS = ['B', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'];;
+
+export function formatBytes(bytes: number | null, decimals: number) {
+    if (bytes == null) {
+        return '-';
+    }
+
+    if (bytes == 0) {
+        return '0 B';
+    }
+
+    let dm = decimals || 2;
+    let i = Math.floor(Math.log(bytes) / Math.log(K));
+
+    return parseFloat((bytes / Math.pow(K, i)).toFixed(dm)) + ' ' + UNITS[i];
+}
```

Comment:
Nice! Tomorrow I'll try the native format function, it should support bytes from what I see in the doc:

https://next-intl-docs.vercel.app/docs/usage/numbers#formatting-plain-numbers

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat#options

https://tc39.es/ecma402/#table-sanctioned-single-unit-identifiers

Reply:
Ok, that would be much better than this that I found on a Github gist :rofl: